sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdk.java

```diff
@@ -60,8 +66,23 @@ static AutoConfiguredOpenTelemetrySdk create(
   /** Returns the {@link Resource} that was auto-configured. */
   abstract Resource getResource();
 
-  /** Returns the {@link ConfigProperties} used for auto-configuration. */
+  /**
+   * Returns the {@link ConfigProperties} used for auto-configuration, or {@code null} if file
+   * configuration was used.
+   *
+   * @see #getStructuredConfig()
+   */
+  @Nullable
   abstract ConfigProperties getConfig();
 
+  /**
+   * Returns the {@link StructuredConfigProperties} used for auto-configuration, or {@code null} if
+   * file configuration was not used.
+   *
+   * @see #getConfig()
+   */
+  @Nullable
+  abstract StructuredConfigProperties getStructuredConfig();
```

Comment:
This new method allows callers to determine if autoconfiguration was driven by file configuration or by env vars / system properties. Either this or `getConfig` will return null. The other will be non-null.

If file configuration is used, `getStructuredConfig` represents the full YAML config file.

Reply:
This is kinda the only unfortunate downside of this approach that I can find thus far. It's a small drag that users of the autoconfigured sdk now have to be aware or otherwise check which config was used to get the data they need.

It feels like there is room to flatten the structured config into `ConfigProperties` and also to objectify the `ConfigProperties` into a structured thing...but we can have that conversation another time.

Reply:
>  It's a small drag that users of the autoconfigured sdk now have to be aware or otherwise check which config was used to get the data they need.

I would characterize it slightly differently: Providers of custom components like exporters and samplers need to be aware that somebody can use file based config or flat system property / env var based config, and provide different SPI implementations for each. This is frustrating, but seems to be the inevitable result of otel getting as far as it did without a proper configuration story. As I mentioned in the PR description, I started with the opposite approach and tried to have a single unified `ConfigProperties` for both configuration routes. I've concluded while this is simpler from an API standpoint, its more confusing and brittle for users, since even though we can do some gymnastics to make `ConfigProperties` work for both cases, the data presented to SPI implementers will be different (i.e. the property names and structure and semantics will be different for flat env vars than in structured config). So there appears to be no getting around confronting the user with this difference. Might as well make it as explicit and obvious as possible.